src/ring_theory/hahn_series.lean

```diff
@@ -583,6 +628,24 @@ lemma C_one : C (1 : R) = (1 : hahn_series Γ R) := C.map_one
 lemma C_mul_eq_smul {r : R} {x : hahn_series Γ R} : C r * x = r • x :=
 single_zero_mul_eq_smul
 
+lemma C_injective : function.injective (C : R → hahn_series Γ R) :=
+begin
+  intros r s rs,
+  rw [ext_iff, function.funext_iff] at rs,
+  have h := rs 0,
+  rwa [C_apply, single_coeff_same, C_apply, single_coeff_same] at h,
+end
+
+lemma C_ne_zero {r : R} (h : r ≠ 0) : (C r : hahn_series Γ R) ≠ 0 :=
+begin
+  contrapose! h,
+  rw ← C_zero at h,
+  exact C_injective h,
+end
+
+lemma order_C {r : R} (h : r ≠ 0) : order (C r : hahn_series Γ R) (C_ne_zero h) = 0 :=
+order_single h
+
```

Comment:
Could you please add simp-lemmas for the order of `0` and `1`?

Reply:
I could alternately define order to be 0 when the value type has zero, and not define order otherwise.

Reply:
I sorry, I forgot again about that hypothesis...
I actually think that making `order` a total function will be helpful in practice. So if you could do that, that would be great.

Reply:
You're right, it's much better!